### PR TITLE
[WIP] Add NOT NULL constraints to repository

### PR DESCRIPTION
**Pull request by ethantkoenig**
#962 

I'm having trouble with my migration; right now, it does not actually add `NOT NULL` constraints to the existing columns of the `repository` table. If someone who knows what they're doing could point me in the right direction, I would appreciate it! :smile:

I've tested locally using Postgres 9.5.5

### CODE REVIEW

File: models/migrations/v17.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+	if _, err := sess.Exec("UPDATE `repository` SET num_pulls = 0 WHERE num_pulls IS NULL"); err != nil {
+		return err
+	}
+
+	if _, err := sess.Exec("UPDATE `repository` SET num_closed_pulls = 0 WHERE num_closed_pulls IS NULL"); err != nil {
+		return err
+	}
+
+	if err := sess.Sync2(new(RepoV17)); err != nil {
```
Comment (ethantkoenig) [inline models/migrations/v17.go:63]: I had assumed that calling `Sync2(..)` would add the new `NOT NULL` constraints to the table, but it doesn't seem to be working.

Comment (lunny) [reply models/migrations/v17.go:63]: No. It will not.

Comment (ethantkoenig) [reply models/migrations/v17.go:63]: So what should I do?

Comment (lunny) [reply models/migrations/v17.go:63]: I think you have to write SQL yourself.

**Comment by ethantkoenig** [comment]
This will be more annoying than I had initially thought, since SQLite and TiDB don't support adding constraints to existing columns. I'll put this on hold for now.

**Pull request closed without merging**